smst.go

```diff
@@ -20,37 +25,39 @@ type SMST struct {
 func NewSparseMerkleSumTrie(
 	nodes kvstore.MapStore,
 	hasher hash.Hash,
-	options ...Option,
+	options ...TrieSpecOption,
 ) *SMST {
+	trieSpec := newTrieSpec(hasher, true)
+	for _, option := range options {
+		option(&trieSpec)
+	}
+
+	// Initialize a non-sum SMT and modify it to have a nil value hasher
+	// TODO_UPNEXT(@Olshansk): Understand the purpose of the nilValueHasher and
```

Comment:
This is because the SMST pre-hashes its paths and thus would result in double path hashing as it is a wrapper around the SMT. The SMST uses its own path hashing logic to account for the sum so the SMT underneath needs a nil path hasher and just uses the paths provided.